Mathlib/Data/Nat/Basic.lean

```diff
@@ -945,3 +945,6 @@ instance decidableExistsLe {P : ℕ → Prop} [DecidablePred P] :
 #align nat.decidable_exists_le Nat.decidableExistsLe
 
 end Nat
+
+/-- A formal statement of Fermat's Last Theorem. -/
+def FermatsLastTheorem : Prop := ∀ a b c n : ℕ, 2 < n ∧ 0 < a ∧ 0 < b → a^n + b^n ≠ c^n
```

Comment:
I think the following will satisfy @ocfnash for the entire function case.
```suggestion
def FermatLastTheorem (α : Type*) [Semiring α] (n : ℕ) : Prop := ∀ a b c : α, a ≠ 0 → b ≠ 0 → a^n + b^n ≠ c^n
```

Reply:
I do think that moving `n` before `a b c` is definitely the right call, since then you can prove FLT for specific exponents

Reply:
Also, the statement is not quite correct as stated, since if n is odd then there are solutions with c=0.

Reply:
Well spotted!